spring-kafka/src/main/java/org/springframework/kafka/retrytopic/FixedDelayStrategy.java

```diff
@@ -22,8 +22,10 @@
  *
  * @author Tomaz Fernandes
  * @since 2.7
+ * @deprecated in a future release, will be replaced by {@link SameIntervalTopicReuseStrategy}.
```

Comment:
We can simply say: `deprecated in favor of {@link SameIntervalTopicReuseStrategy}`

Reply:
Ok I will proceed with the changes

Reply:
> We can simply say: `deprecated in favor of {@link SameIntervalTopicReuseStrategy}`

Ok.

I do not think we can say that.
Previously, I had said that I was having hard time trying to condense (internally at `RetryTopicConfigurationBuilder` and `DestinationTopicPropertiesFactory`) both configurations, in order for these components to work only with the `SameIntervalTopicReuseStrategy` configuration. So that, when a method like `RetryTopicConfigurationBuilder.useSingleTopicForFixedDelays()` was called, internally this would actually configure the `SameIntervalTopicReuseStrategy`.

But then, we aggreed on keeping both configurations. Just marking the old configurations as deprecated, aiming to remove them in the future (when `DestinationTopicPropertiesFactory` should be changed).

So we cannot say to make use of `SameIntervalTopicReuseStrategy`, because right now, for Fixed delay topics, `FixedDelayStrategy` still must be used.

Reply:
And to recap... when trying to condense the configs, thus allowing new code to configure single fixed delay topic using the new methods like `useSingleTopicForSameIntervals()`, the real problem is in `RetryableTopicAnnotationProcessor` and how it would interact correctly with the builder.

Like I explained [here](https://github.com/spring-projects/spring-kafka/pull/2497#issuecomment-1418201556).

Reply:
Sorry, I've read your comment before, I read it again now, and I'm having a hard time understanding why this is so complicated.

In my head, we can just leave both methods / parameters in all classes you mentioned, deprecating the old ones, and solve any ambiguity in the `DestinationTopicPropertiesFactory` class, by the rules I mentioned in the other comment I made earlier:

> Both enums default to MULTIPLE_TOPICS
For the new logic, only SameIntervalTopicReuseStrategy is looked up, so no problems there.
For existing logic, we can look at both enum values - if any of them is SINGLE_TOPIC, single topic it is.

Can you please help me understand why wouldn't this work? Thanks.

Reply:
Or being more specific:

- For fixed delays, we can look at both the old and new enums - whichever has `SINGLE_TOPIC` has precedence.
- For exponential (non-fixed) delays we only look at the new enum.

In your example:
> At first, one case I can think of is existing code like where the developer declared FixedDelayStrategy.SINGLE_TOPIC, but is in fact using an exponential backoff

For this case, we'd assume `SINGLE_TOPIC` for fixed delays, just as it is now. For exponential delays the new enum would be `MULTIPLE_TOPICS` so wouldn't change functionality. If the user sets the new enum for `SINGLE_TOPIC`, the new enum should take precedence and impact both the fixed delay and exponential delay variants.

Also, we don't really need to take care of all combinations. E.g. if the user explicitly sets the old enum to `MULTIPLE` and the new one to `SINGLE`, that's a misconfiguration (using both a deprecated and the new configuration at the same time) and we can just assume `SINGLE`. At most we'd need to document this in the javadocs and / or documentation.

Does this make sense? Thanks.